.github/workflows/ci-deep.yml

```diff
@@ -68,12 +68,24 @@ jobs:
     steps:
       - name: "Check out the repo"
         uses: "actions/checkout@v3"
-        with:
-          submodules: "recursive"
 
       - name: "Install Foundry"
         uses: "foundry-rs/foundry-toolchain@v1"
 
+      - name: "Install Pnpm"
+        uses: "pnpm/action-setup@v2"
+        with:
+          version: "8"
+
+      - name: "Install Node.js"
+        uses: "actions/setup-node@v3"
+        with:
+          cache: "pnpm"
+          node-version: "lts/*"
+
+      - name: "Install the Node.js dependencies"
+        run: "pnpm install"
+
```

Comment:
We need to find a way to shorten this syntax - it's too verbose.

Related: https://github.com/sablier-labs/v2-periphery/issues/152

Reply:
I've managed to cache the `node_modules` dir and it seems to work expect the coverage, which i don't understand

Reply:
I wasn't referring to the `run: "pnpm install"` step. I was referring to all of these separate steps - it would be great if we could have a shared job. See the linked issue.

> I've managed to cache the node_modules dir and re-use it

That's not what that `cache:` key does. It only caches global packages. See:

- https://github.com/actions/setup-node?tab=readme-ov-file#caching-global-packages-data
- https://github.com/actions/setup-node/issues/612

I will bring back that key

Reply:
Oh nevermind I looked at the code now and I see what you did.

Great idea to cache the `node_modules`!